src/libraries/Microsoft.Extensions.Logging.Console/src/SimpleConsoleFormatter.cs

```diff
@@ -0,0 +1,247 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class SimpleConsoleFormatter : ConsoleFormatter, IDisposable
+    {
+        [ThreadStatic]
+        private static StringBuilder _logBuilder;
+        private const string LoglevelPadding = ": ";
+        private static readonly string _messagePadding = new string(' ', GetLogLevelString(LogLevel.Information).Length + LoglevelPadding.Length);
+        private static readonly string _newLineWithMessagePadding = Environment.NewLine + _messagePadding;
+        private IDisposable _optionsReloadToken;
+
+
+        public SimpleConsoleFormatter(IOptionsMonitor<SimpleConsoleFormatterOptions> options)
+        : base (ConsoleFormatterNames.Simple)
+        {
+            FormatterOptions = options.CurrentValue;
+            ReloadLoggerOptions(options.CurrentValue);
+            _optionsReloadToken = options.OnChange(ReloadLoggerOptions);
+        }
+
+        private void ReloadLoggerOptions(SimpleConsoleFormatterOptions options)
+        {
+            FormatterOptions = options;
+        }
+
+        public void Dispose()
+        {
+            _optionsReloadToken?.Dispose();
+        }
+
+        internal SimpleConsoleFormatterOptions FormatterOptions { get; set; }
+
+        public override void Write<TState>(in LogEntry<TState> logEntry, IExternalScopeProvider scopeProvider, TextWriter textWriter)
+        {
+            LogLevel logLevel = logEntry.LogLevel;
+            ConsoleColors logLevelColors = GetLogLevelConsoleColors(logLevel);
+            string logLevelString = GetLogLevelString(logLevel);
+
+            string timestamp = null;
+            string timestampFormat = FormatterOptions.TimestampFormat;
+            if (timestampFormat != null)
+            {
+                DateTime dateTime = GetCurrentDateTime();
+                timestamp = dateTime.ToString(timestampFormat);
+            }
+            if (timestamp != null)
+            {
+                textWriter.Write(FormatterOptions.SingleLine ? timestamp + ' ' : timestamp);
+            }
+
+            if (logLevelString != null)
+            {
+                textWriter.WriteColoredMessage(logLevelString, logLevelColors, FormatterOptions.DisableColors);
+            }
+
+            StringBuilder logBuilder = _logBuilder;
```

Comment:
>logBuilder [](start = 26, length = 10)

could you please explain more about storing the builder in a local variable? is this mainly to avoid the re-entry of this method from other threads?

Reply:
this one I am actually keeping as-is how it was prior to this PR.

https://github.com/dotnet/runtime/blob/master/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLogger.cs#L63-L69

Reply:
@davidfowl  any idea why that?

---
In reply to: [451215996](https://github.com/dotnet/runtime/pull/38616#discussion_r451215996) [](ancestors = 451215996)

Reply:
removed the string builders from all formatters